client/src/main/java/io/pravega/client/tables/TableSegment.java

```diff
@@ -0,0 +1,172 @@
+     * keys that have a value in the index. All other keys will not be included.
+     */
+    CompletableFuture<Map<KeyT, GetResult<ValueT>>> get(Collection<KeyT> keys);
+
+    /**
+     * Creates and registers a {@link KeyUpdateListener} for all updates to this {@link TableSegment}, subject to the given
+     * {@link KeyUpdateFilter}.
+     *
+     * @param filter   A {@link KeyUpdateFilter} that will specify which Keys should the {@link KeyUpdateListener} listen to.
+     * @param executor An {@link Executor} that will be used to invoke all callbacks on.
+     * @return A CompletableFuture that, when completed, will contain an {@link KeyUpdateListener}.
+     */
+    CompletableFuture<KeyUpdateListener<KeyT, ValueT>> createListener(KeyUpdateFilter<KeyT> filter, Executor executor);
+
+    /**
+     * Creates a new Iterator over all the Keys in the Table Segment.
+     *
+     * @param continuationToken An {@link IteratorState} that represents a continuation token that can be used to resume
+     *                          a previously interrupted iteration. This can be obtained by invoking
+     *                          {@link IteratorItem#getContinuationToken()}. A null value will create an iterator that
+     *                          lists all keys.
+     * @return A CompletableFuture that, when completed, will return an {@link AsyncIterator} that can be used to iterate
+     * over all the Keys in this Table Segment.
+     */
+    CompletableFuture<AsyncIterator<IteratorItem<TableKey<KeyT>>>> keyIterator(IteratorState continuationToken);
+
+    /**
+     * Creates a new Iterator over all the Entries in the Table Segment.
+     *
+     * @param continuationToken An {@link IteratorState} that represents a continuation token that can be used to resume
+     *                          a previously interrupted iteration. This can be obtained by invoking
+     *                          {@link IteratorItem#getContinuationToken()}. A null value will create an iterator that
+     *                          lists all Entries.
+     * @return A CompletableFuture that, when completed, will return an {@link AsyncIterator} that can be used to iterate
+     * over all the Entries in this Table Segment.
+     */
+    CompletableFuture<AsyncIterator<IteratorItem<TableEntry<KeyT, ValueT>>>> entryIterator(IteratorState continuationToken);
+
+    @Override
+    void close();
+
+    /**
+     * An iteration result item returned by {@link AsyncIterator} when invoking {@link #entryIterator(IteratorState)} or
+     * {@link #keyIterator(IteratorState)}.
+     *
+     * @param <T> Iterator Item type..
+     */
+    @Data
+    class IteratorItem<T> {
+        /**
+         * Gets an {@link IteratorState} that can be used to reinvoke {@link TableSegment#entryIterator(IteratorState)} or
+         * {@link TableSegment#keyIterator(IteratorState)}if a previous iteration has been interrupted (by losing the
```

Comment:
For my own understanding, could you elaborate on what this is referring to?

Reply:
For a table with 100 million keys, there is no way you can fetch all of them at once, hence you need an iterator. 
When you iterate over so many keys, it will take a long time, during which your process may crash or some other unfortunate event may happen. If you do not want to completely restart your iteration, providing such an `IteratorState` (contents TBD) will help you resume the iteration from where it left off.